Add react-router to restricted imports

lfryemason opened the pull request:
`react-router` and `react-router-dom` have identical names for many functions, but we only want to use `react-router-dom` for all future projects. Add `react-router` to the no-restricted-imports list means that there is a nice error message when I accidentally import react-router into vscode

davidsbailey reviewed (approved):
Hooray for linting! Thanks for doing this Liam

Nokondi reviewed (approved):
👍

lfryemason commented:
Commit history may look weird because I accidentally removed the commit that was approved. Based on the branch name, I assumed that I had a different branch for the lint `no-restricted-imports` PR and this branch was mistakenly branched off of that branch. I therefore based the branch off staging and then starting working on my next task. Turns out, that there was no separate branch for the linting rule and I had deleted the commit with the linting rule.

I fixed this by:
* resetting the next commit from my next task and put it into a different branch
* Cherry-picking the commit that has the linting rule.
* Force pushing with the cherry-picked commit and the commit with the next task removed.
